### ISSUE TITLE
Plugin does not load in KeePass2 2.48.1 Portable on Ubuntu 20.04 anymore

### ISSUE THREAD

**Issue by meyca** [tiuub/KeeOtp2#29]
Up to version 1.5.3 the plugin does load. 1.5.4 and 1.5.5 do not.

`$ mono ~/Applications/Keepass2/KeePass.exe --debug
`

![image](https://user-images.githubusercontent.com/408161/122873161-b779c080-d331-11eb-9051-96d9453e14eb.png)

![image](https://user-images.githubusercontent.com/408161/122873213-c791a000-d331-11eb-90ce-ce74d333571f.png)

```
$ cat /tmp/tmp4fba55.tmp 
$ 

```
The file is empty.

Package "mono-complete" is installed (as in Keepass-Error-dialogue suggested when running without "--debug").
```
$ sudo apt list --installed | grep mono
[sudo] Passwort für mm: 

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

ca-certificates-mono/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
fonts-noto-mono/focal-updates,focal-updates,now 20200323-1build1~ubuntu20.04.1 all  [Installiert,automatisch]
fonts-tlwg-mono-ttf/focal,focal,now 1:0.7.1-3 all  [Installiert,automatisch]
fonts-tlwg-mono/focal,focal,now 1:0.7.1-3 all  [Installiert,automatisch]
libmono-2.0-1/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
libmono-2.0-dev/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
libmono-accessibility4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-btls-interface4.0-cil/focal,now 6.8.0.105+dfsg-2 amd64  [installiert]
libmono-cairo4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-cecil-private-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-cil-dev/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-codecontracts4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-compilerservices-symbolwriter4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-corlib4.5-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-cscompmgd0.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-csharp4.0c-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-custommarshalers4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-data-tds4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-db2-1.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-debugger-soft4.0a-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-http4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-i18n-cjk4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-i18n-mideast4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-i18n-other4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-i18n-rare4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-i18n-west4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-i18n4.0-all/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-i18n4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-ldap4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-management4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-messaging-rabbitmq4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-messaging4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-build-engine4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-build-framework4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-build-tasks-v4.0-4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-build-utilities-v4.0-4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-build4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-csharp4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-visualc10.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-microsoft-web-infrastructure1.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-oracle4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-parallel4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-peapi4.0a-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-posix4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-profiler/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
libmono-rabbitmq4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-relaxng4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-security4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-sharpzip4.84-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-simd4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-smdiagnostics0.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-sqlite4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-componentmodel-composition4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-componentmodel-dataannotations4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-configuration-install4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-configuration4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-core4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-data-datasetextensions4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-data-entity4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-data-linq4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-data-services-client4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-data-services4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-data4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-deployment4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-design4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-drawing-design4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-drawing4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-dynamic4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-enterpriseservices4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-identitymodel-selectors4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-identitymodel4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-io-compression-filesystem4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-io-compression4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-json-microsoft4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-json4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-ldap-protocols4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-ldap4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-management4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-messaging4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-net-http-formatting4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-net-http-webrequest4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-net-http4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-net4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-numerics-vectors4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-numerics4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-core2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-debugger2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-experimental2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-interfaces2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-linq2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-observable-aliases0.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-platformservices2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-providers2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-runtime-remoting2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-windows-forms2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reactive-windows-threading2.2-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-reflection-context4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-runtime-caching4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-runtime-durableinstancing4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-runtime-serialization-formatters-soap4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-runtime-serialization4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-runtime4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-security4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-servicemodel-activation4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-servicemodel-discovery4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-servicemodel-internals0.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-servicemodel-routing4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-servicemodel-web4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-servicemodel4.0a-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-serviceprocess4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-threading-tasks-dataflow4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-transactions4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-abstractions4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-applicationservices4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-dynamicdata4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-extensions-design4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-extensions4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-http-selfhost4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-http-webhost4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-http4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-mobile4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-mvc3.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-razor2.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-regularexpressions4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-routing4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-services4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-webpages-deployment2.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-webpages-razor2.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web-webpages2.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-web4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-windows-forms-datavisualization4.0a-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-windows-forms4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [installiert]
libmono-system-windows4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-workflow-activities4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-workflow-componentmodel4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-workflow-runtime4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-xaml4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-xml-linq4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-xml-serialization4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system-xml4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-system4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-tasklets4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-webbrowser4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-webmatrix-data4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-windowsbase4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmono-xbuild-tasks4.0-cil/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
libmonoboehm-2.0-1/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
libmonosgen-2.0-1/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
libmonosgen-2.0-dev/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
mono-4.0-gac/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
mono-4.0-service/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
mono-complete/focal,now 6.8.0.105+dfsg-2 amd64  [installiert]
mono-csharp-shell/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
mono-devel/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
mono-gac/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
mono-mcs/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
mono-runtime-common/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
mono-runtime-sgen/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
mono-runtime/focal,now 6.8.0.105+dfsg-2 amd64  [installiert]
mono-utils/focal,now 6.8.0.105+dfsg-2 amd64  [Installiert,automatisch]
mono-xbuild/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
mono-xsp4-base/focal,focal,now 4.2-2.1 all  [Installiert,automatisch]
mono-xsp4/focal,focal,now 4.2-2.1 all  [Installiert,automatisch]
monodoc-base/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
monodoc-http/focal,focal,now 4.2-3 all  [Installiert,automatisch]
monodoc-manual/focal,focal,now 6.8.0.105+dfsg-2 all  [Installiert,automatisch]
ubuntu-mono/focal,focal,now 19.04-0ubuntu3 all  [Installiert,automatisch]

```

**Comment by tiuub** [comment]
Hi @meyca,

thank you for your submission. I recently noticed, that KeeOtp2 is broken for Linux. At first, I fixed some other bugs, but I will get my eyes on this issue now. Please stay patient.

Kind regards

**Comment by andoo** [comment]
I really love this plugin and would love to use it on my linux machine too :-)

For me, the generated tmp-file was not empty. It seems to be a problem with the filesystem. It says:

> error CS2001: Source file `/run/user/1000/KeePass/2D68BF58DA26DE4EA017D1D6DAEA63EA/KeeOtp2/Forms/Settings.designer.cs' could not be found
> error CS2001: Source file `/run/user/1000/KeePass/2D68BF58DA26DE4EA017D1D6DAEA63EA/KeeOtp2/Forms/Troubleshooting.designer.cs' could not be found

But the files are named _Settings.Designer.cs_ and _Troubleshooting.Designer.cs_ (uppercase D instead of lowercase d).

**Comment by japrudhomme** [comment]
I am having this same issue. I am happy to help with testing a fix if needed. I am on Linux Mint 20.2 Uma base: Ubuntu 20.04 focal

**Comment by tiuub** [comment]
Finally, I could get a fix for this issue. 

I had tested it on Fedora. Please give me some feedback, if it is working on your system too.

Kind regards
tiuub

**Comment by meyca** [comment]
confirm fixed with KeePass 2.49 and Plugin 1.5.7.0 on Ubuntu 20.04.

Thank You very much, @tiuub.
